wallet-new/src/Cardano/Wallet/WalletLayer/Legacy/Settings.hs

```diff
@@ -0,0 +1,29 @@
+module Cardano.Wallet.WalletLayer.Legacy.Settings
+    ( getSettings
+    ) where
+
+import           Universum
+
+import           Cardano.Wallet.API.V1.Migration
+import           Cardano.Wallet.API.V1.Types as V1
+import qualified Data.Text as T
+import           Paths_cardano_sl_wallet_new (version)
+
+import           Pos.Update.Configuration (curSoftwareVersion)
+import           Pos.Util.CompileInfo (compileInfo, ctiGitRevision)
+import           Pos.Wallet.WalletMode (MonadBlockchainInfo, blockchainSlotDuration)
+
+
+-- Returns the @static@ settings for this wallet node,
+-- like the slot duration or the current 'SoftwareVersion'.
+getSettings
+    :: forall m. (HasConfigurations, HasCompileInfo, MonadBlockchainInfo m)
+    => m NodeSettings
+getSettings = do
+    settings <- NodeSettings <$> (V1.mkSlotDuration . fromIntegral <$> blockchainSlotDuration)
+                             <*> pure (V1 curSoftwareVersion)
+                             <*> pure version
+                             <*> pure (T.replace "\n" mempty $ ctiGitRevision compileInfo)
+
+    pure settings
```

Comment:
the pattern

```haskell
x <- foo bar
pure x
```
can be simplified to

```haskell
foo bar
```

though, more generally, I think this could be:

```
getSettings = do
    duration <- V1.mkSlotDuration . fromIntegral <$> blockahinSlotDuration
    pure $
        NodeSettings duration (V1 curSoftwareVersion) version (T.replace "\n" mempty $ ctiGitRevision compileInfo) 
```

to make it more clear where the effects are happening

Reply:
This could be better, but for now, copied the existing code.